main/SelectBox/index.tsx

```diff
@@ -0,0 +1,190 @@
+import {
+  Animated,
+  Easing,
+  Platform,
+  StyleProp,
+  TextStyle,
+  View,
+  ViewStyle,
+} from 'react-native';
+import {DoobooTheme, light} from '../theme';
+import React, {FC, ReactElement, useEffect, useRef, useState} from 'react';
+
+import {Icon} from '../Icon';
+import {TouchableOpacity} from 'react-native-gesture-handler';
+import {Typography} from '../Typography';
+import styled from '@emotion/native';
+import {withTheme} from '@emotion/react';
+
+const Title = styled.View`
+  width: 200px;
+  height: 30px;
+  border-width: 1px;
+
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+`;
+
+const Item = styled.View`
+  height: 30px;
+  width: 200px;
+  border-bottom-width: 1px;
+  border-left-width: 1px;
+  border-right-width: 1px;
+
+  justify-content: center;
+  align-items: center;
+`;
+
+type Styles = {
+  titleContainer?: StyleProp<ViewStyle>;
+  titleText?: StyleProp<TextStyle>;
+  rightElementContainer?: StyleProp<ViewStyle>;
+  itemContainer?: StyleProp<ViewStyle>;
+  itemText?: StyleProp<TextStyle>;
+};
+interface ItemCompProps {
+  value: string;
+  order: number;
+  styles?: Styles;
+  setIsOpened: (value: boolean) => void;
+  itemActiveOpacity: number;
+  onPress?: (i: number) => void;
+}
+
+const ItemComp: FC<ItemCompProps & {theme: DoobooTheme}> = ({
+  value,
+  order,
+  styles,
+  theme,
+  setIsOpened,
+  itemActiveOpacity,
+  onPress,
+}) => {
+  const handlePress = (): void => {
+    onPress?.(order);
+    setIsOpened(false);
+  };
+
+  return (
+    <TouchableOpacity onPress={handlePress} activeOpacity={itemActiveOpacity}>
+      <Item
+        style={[
+          {
+            borderColor: theme.primary,
+            backgroundColor: theme.textContrast,
+          },
+          styles?.itemContainer,
+        ]}>
+        <Typography.Body2 style={styles?.itemText}>{value}</Typography.Body2>
+      </Item>
+    </TouchableOpacity>
+  );
+};
+
+export interface Props {
+  data: string[];
+  onPress?: (i: number) => void;
+  selectedIndex?: number;
+  style?: StyleProp<ViewStyle>;
+  styles?: Styles;
+  rotateDuration?: number;
+  titleActiveOpacity?: number;
+  itemActiveOpacity?: number;
+  shouldRotate?: boolean;
```

Comment:
We can't think of this as an `animated` props at a glance. Suggest you to rename this